pkg/build/strategies/layered/layered.go

```diff
@@ -49,6 +50,22 @@ func getDestination(config *api.Config) string {
 	return destination
 }
 
+// if dir is empty, or not accessible, we say scripts are not provided
+func scriptsProvided(name string) bool {
+	f, err := os.Open(name)
+	if err != nil {
+		return false
+	}
+	defer f.Close()
+
+	_, err = f.Readdir(1)
+	if err == io.EOF {
+		// empty - means no scripts provided
```

Comment:
You just re-implemented ioutil.Readdir here :-) See: https://golang.org/src/io/ioutil/ioutil.go?s=3208:3259#L90

This should be utility function, something like:

```go
func checkEmptyDir(name string) {
  items, err := ioutil.ReadDir(name)
  if !isNotFound(err) {
    glog.Warningf("Unable to access directory %q: %v", name, err)
  }
  return !(err != nil || len(items) == 0)
}
```

Also I think we should report the errors here as the script might be provided but for some reason (permissions, corrupted filesystem, etc) we can't get them. So I would also log the error when then error is other than NotFound